### DISCUSSION TITLE
I have a problem implementing search function in my laravel Application

### DISCUSSION THREAD

**Discussion by ghost** [community/community#41318] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I need help adding a search function to my Laravel Application. The whole issue has to do with Passing the result in the Route and presenting it in the view.

Code below. I also want to display the results in the same views/students.blade.php view
```
<form action="{{URL::to('/students'}} method="get">
@cstf
  <input type="search" name="search" placeholder="Search Students">
<input type="submit" value="Search Students">
</form>
```

//inside StudentController

```
 use App\Models\Students;
class StudentController extends Controller
{
  public function search(Request $request){
  $students = Students::where('name','LIKE','%'.$request->search.'%')->get();
  return view('students',['students' => $students]);
 }
}
```

//inside web Routes

```
use App\Http\Controllers\StudentController;

 Route::get('/students',function(){
 return view('students');
});

Route::get('/students',[StudentController::class,'search']);
```

//inside views/students.blade.php

```
@foreach($students as $tudent)
   {{$student->id}}
  {{$student->name}}
@endforeach
```
The error in view is: undefined $students.
How do I solve this issue?
Thanks

**Comment by jfullstackdev** [comment, marked as answer]
take a look closely at your `blade.php`, is that a typo ? 

![image](https://user-images.githubusercontent.com/47092464/206842935-dddf2a46-f4c5-4b70-b7a5-0a80d3730437.png)

you have variable `$tudent` not `$student`, so maybe the undefined variable is `$student` not `$students` because in the for loop, it's `$tudent`

if it is `$students`, make sure you include that Entity to be recognized

you can `var_dump` then `exit` to see the variables whether it's really empty or undefined, just start with `@php` then end it `@endphp`, that's always the best way to check variables

**Reply by ghost** [reply]
I think you're right. I got it working. I just need to avoid longer words. Shorter ones make it easy. No typos
